File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/CreateFlagLinkExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class CreateFlagLinkExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey = new Dictionary<string, string> {["ApiKey"] = "YOUR_API_KEY"};

        var flagLinkPost = new FlagLinkPost(
            key: "flag-link-key-123abc",
            deepLink: "https://example.com/archives/123123123",
            title: "Example link title",
            description: "Example link description"
        );

        try
        {
            var response = new FlagLinksBetaApi(config).CreateFlagLink(
                projectKey: null,
                featureFlagKey: null,
                flagLinkPost: flagLinkPost
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling FlagLinksBetaApi#CreateFlagLink: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/CreateFlagLinkExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const flagLinkPost: api.FlagLinkPost = {
  key: "flag-link-key-123abc",
  deepLink: "https://example.com/archives/123123123",
  title: "Example link title",
  description: "Example link description",
};

new api.FlagLinksBetaApi(configuration).createFlagLink(
  "projectKey_string", // projectKey
  "featureFlagKey_string", // featureFlagKey
  flagLinkPost,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling FlagLinksBetaApi#createFlagLink:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
